foundation/blockchain/state/network.go

```diff
@@ -36,26 +39,39 @@ func (s *State) NetSendBlockToPeers(block database.Block) error {
 }
 
 // NetSendTxToPeers shares a new block transaction with the known peers.
-func (s *State) NetSendTxToPeers(tx database.BlockTx) {
+func (s *State) NetSendTxToPeers(ctx context.Context, tx database.BlockTx) {
```

Comment:
I hear what you are saying, but this is really not necessary. In a production version you would want to send this to your local group of peers quickly, that is true. Adding the concurrency here only makes the code harder to read and comprehend for new people. This is running in a goroutine, just these calls are serialized.

Reply:
Also, look at how I applied concurrency. These functions perform the work they need to and the concurrency is coded outside these functions. Look at worker.go.  You really want to separate these things. For the POW code, that required extra concurrency, but notice again the concurrency code does it's thing and the functions that code calls don't know anything about how concurrency.

Reply:
I think I understand what you're saying.
runPowOperation() & runPoaOperation() are responsible for concurrency as opposed to powOperations() & poaOperations(), as they are listening to the cancelMining channel.

The difference, in my opinion, is that NetSendTxToPeers() does not listen to any "cancelMining" equivalent. It is a blocking call, that may block forever as things are right now, and you won't be able to cancel it (hence the new goroutine - to be able to cancel NetSendTxToPeers).

But anyway I completely understand your point as this adding extra complexity, which can be avoided in a reference implementation.

Thanks for the review!